pkg/chunk/chunk_store.go

```diff
@@ -281,10 +281,11 @@ func (c *store) validateQueryTimeRange(ctx context.Context, userID string, from
 		}
 	}
 
-	if through.After(now.Add(5 * time.Minute)) {
+	maxFutureThrough := now.Add(5 * time.Minute)
+	if through.After(maxFutureThrough) {
 		// time-span end is in future ... regard as legal
-		level.Info(log).Log("msg", "adjusting end timerange from future to now", "old_through", through, "new_through", now)
-		*through = now // Avoid processing future part - otherwise some schemas could fail with eg non-existent table gripes
+		level.Info(log).Log("msg", "adjusting end timerange from future to now", "old_through", through, "new_through", maxFutureThrough)
+		*through = maxFutureThrough // Avoid processing future part - otherwise some schemas could fail with eg non-existent table gripes
```

Comment:
There's no guarantee that after this operation `from` < `through`. If we specify `from: now().Add(10min)` and `through: now().Add(15min)`, it'd would violate the guarantee we're erroring on earlier (that `from` < `through`). 

It probably needs another check/error condition here like `errors.New("Adjusted end time would be before start time")`